pennylane/workflow/execution.py

```diff
@@ -259,28 +271,31 @@ def _make_inner_execute(
 
     if isinstance(device, qml.Device):
         device_execution = set_shots(device, override_shots)(device.batch_execute)
-
     else:
         device_execution = partial(device.execute, execution_config=execution_config)
 
-    cached_device_execution = qml.workflow.cache_execute(
-        device_execution, cache, return_tuple=False
-    )
-
     def inner_execute(tapes: Sequence[QuantumTape], **_) -> ResultBatch:
         """Execution that occurs within a machine learning framework boundary.
 
         Closure Variables:
             expand_fn (Callable[[QuantumTape], QuantumTape]): A device preprocessing step
             numpy_only (bool): whether or not to convert the data to numpy or leave as is
-            cached_device_execution (Callable[[Sequence[QuantumTape]], ResultBatch])
-
+            device_execution (Callable[[Sequence[QuantumTape]], ResultBatch])
+            cache (None | MutableMapping): The cache to use. If ``None``, caching will not occur.
         """
+        transform_program = qml.transforms.core.TransformProgram()
+
+        if cache is not None:
+            transform_program.add_transform(_cache_transform, cache=cache)
+
+        # TODO: Apply expand_fn() and convert_to_numpy_parameters() as transforms.
         if expand_fn:
             tapes = tuple(expand_fn(t) for t in tapes)
         if numpy_only:
             tapes = tuple(qml.transforms.convert_to_numpy_parameters(t) for t in tapes)
-        return cached_device_execution(tapes)
+
+        transformed_tapes, transform_post_processing = transform_program(tapes)
+        return transform_post_processing(device_execution(transformed_tapes))

```

Comment:
```suggestion
        transformed_tapes, transform_post_processing = transform_program(tapes)
        if len(transformed_tapes) == 0:
            return transform_post_processing(tuple())
        return transform_post_processing(device_execution(transformed_tapes))
```

This should fix the failures in `tests/interfaces/default_qubit_2_integration/test_execute_default_qubit_2.py` .